src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -756,7 +756,9 @@ public function handleBlockActorData(BlockActorDataPacket $packet) : bool{
 			}
 
 			try{
-				$text = SignText::fromBlob($textBlobTag->getValue());
+				$baseColor = $block->getText()->getBaseColor();
+				$glowing = $block->getText()->isGlowing();
+				$text = SignText::fromBlob($textBlobTag->getValue(), $baseColor, $glowing);
```

Comment:
This allows the player to change the sign properties without the requisite items.

Reply:
I don't understand how this code allows players to change sign properties without items, the code only looks up for the sign which is actually placed on the world, not depending on what the client send.

Reply:
Then why change this at all?

Reply:
The previous code doesn't consider the sign color and glowing state, resulting in the removal of color and glowing states when the player edited the sign.

Reply:
My bad, I misread the code.

The new variables for the formatting should be moved out of the try-catch, as they shouldn't be throwing any exceptions.